python/taichi/lang/matrix.py

```diff
@@ -403,6 +403,16 @@ class Matrix(TaichiOperations):
         0
     """
     _is_taichi_class = True
+    __arrar_priority__ = 1000
+
+    def __array_ufunc__(self, ufunc, method, *args, **kwargs):
```

Comment:
I'm not very familiar with numpy, so could you explain what this function does?

Reply:
It's here: https://numpy.org/neps/nep-0018-array-function-protocol.html

If the user's custom array implements this interface, NumPy will call this function to handle the operations between ndarray and user's custom arrays.